source/game/field/obj/ObjectTwistedWay.hh

```diff
@@ -0,0 +1,103 @@
+    [[nodiscard]] bool checkSphereCachedPartial(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset) override;
+    [[nodiscard]] bool checkSphereCachedPartialPush(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset) override;
+    [[nodiscard]] bool checkSphereCachedFull(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset) override;
+    [[nodiscard]] bool checkSphereCachedFullPush(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset) override;
+
+private:
+    [[nodiscard]] bool checkSpherePartialImpl(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset);
+    [[nodiscard]] bool checkSpherePartialPushImpl(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset);
+    [[nodiscard]] bool checkSphereFullImpl(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset);
+    [[nodiscard]] bool checkSphereFullPushImpl(f32 radius, const EGG::Vector3f &v0,
+            const EGG::Vector3f &v1, KCLTypeMask flags, CollisionInfo *pInfo,
+            KCLTypeMask *pFlagsOut, u32 timeOffset);
+
+    [[nodiscard]] bool checkSphereImpl(f32 radius, const EGG::Vector3f &v0, const EGG::Vector3f &v1,
+            KCLTypeMask flags, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut, u32 timeOffset,
+            bool full, bool push);
+
+    [[nodiscard]] bool FUN_80814270(f32 radius, f32 param_3, const EGG::Vector3f &relativePos,
+            EGG::Vector3f &v0, EGG::Vector3f &fnrm, f32 &dist);
+
+    [[nodiscard]] f32 calcWallHeightOffset(f32 zAxisProgress, s32 phase);
+
+    [[nodiscard]] bool checkSphereWallCollision(f32 heightOffset, f32 radius, s32 phase,
+            const EGG::Vector3f &relativePos, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut,
+            bool full, bool push);
+
+    [[nodiscard]] bool checkSphereRoadCollision(f32 heightOffset, f32 radius,
+            const EGG::Vector3f &relativePos, CollisionInfo *pInfo, KCLTypeMask *pFlagsOut,
+            bool full, bool push);
+
+    f32 m_ac = 1000.0f;
+
+    static constexpr f32 AMPLITUDE = 0.2f;
+    static constexpr f32 WAVINESS = 4.0f;
+    static constexpr s32 PHASE_COUNT = 120;
+
+    static constexpr f32 WIDTH = 2000.0f;
+    static constexpr f32 _B8 = 2000.0f;
```

Comment:
Couple uses of `_B8` still, but if it's the same as width, does it make sense to just use `WIDTH` instead? Or were these left over?